nj2k/src/org/jetbrains/kotlin/nj2k/conversions/ContinueStatementConversion.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.nj2k.conversions
+
+import com.intellij.psi.*
+import org.jetbrains.kotlin.nj2k.*
+import org.jetbrains.kotlin.nj2k.tree.*
+
+// This conversion is for converting continue statements. For example, this adds updaters in converted while loops, and attaches
+// labels named 'loop' to continue statements inner 'when' in loops and their corresponding loops for avoiding syntax errors.
+// If the 'continue' keywords in 'switch' is allowed by the kotlin syntax, ContinueStatementConverter, convertForInStatement and
+// convertWhileStatement in this file can be removed.
+class ContinueStatementConversion(context: NewJ2kConverterContext) : RecursiveApplicableConversionBase(context) {
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        if (element is JKKtConvertedFromForLoopSyntheticWhileStatement) {
+            return recurse(convertSyntheticWhileStatement(element))
+        }
+
+        if (element is JKForInStatement) {
+            convertForInStatement(element)?.also { return it }
+        }
+
+        if (element is JKWhileStatement) {
+            convertWhileStatement(element)?.also { return it }
+        }
+
+        return recurse(element)
+    }
+
+    private fun convertForInStatement(loopStatement: JKForInStatement): JKStatement? {
+        val continueStatementConverter = ContinueStatementConverter(loopStatement, context)
+        val body = continueStatementConverter.apply(loopStatement::body.detached())
+        if (!continueStatementConverter.needLabel || body !is JKStatement) {
+            return null
+        }
+        return JKLabeledExpression(
+            JKForInStatement(
```

Comment:
Instead of recreating `for` statemetns every time you can just update its bodies. This applicable for all three types of loops